apps/dynamo/lib/dynamo/vnode.ex

```diff
@@ -81,6 +105,35 @@ defmodule Vnode do
     end
   end
 
+  defp wait_read_response(key, value, index, context, state, current_read, num_read) do
+    if current_read < num_read do
+      receive do
+        {:ok, ^key, {other_value, other_context}, other_index} ->
+          cond do
+            VClock.compare_vclocks(context, other_context) == :after ->
+              Logger.info("vclock received. Result of comparison: After")
+              # Need to send a put request to the node behind
+              wait_read_response(key, value, index, context, state, current_read+1, num_read)
+            VClock.compare_vclocks(context, other_context) == :before ->
+              Logger.info("vclock received. Result of comparison: Before")
+              index_map = Map.get(state.data, key, %{})
+              index_map = Map.put(index_map, key, {other_value, other_context})
+              state = %{state | data: index_map}
+              wait_read_response(key, other_value, index, other_context, state, current_read+1, num_read)
```

Comment:
Why does `current_read` get incremented even it has receive different value?

Reply:
Just wrote it for now so that the code doesnt get stuck

Reply:
And thinking about it it should increment even if value is different, basically we should return all the available values we get from nodes if their vclocks are concurrent, so it should be fine